pkg/api/addons.go

```diff
@@ -306,8 +306,12 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		Enabled: to.BoolPtr(o.IsAzureCNI() && o.KubernetesConfig.NetworkPolicy != NetworkPolicyCalico),
 		Containers: []KubernetesContainerSpec{
 			{
-				Name:  common.AzureCNINetworkMonitorAddonName,
-				Image: specConfig.AzureCNIImageBase + k8sComponents[common.AzureCNINetworkMonitorAddonName],
+				Name:           common.AzureCNINetworkMonitorAddonName,
+				Image:          specConfig.AzureCNIImageBase + k8sComponents[common.AzureCNINetworkMonitorAddonName],
+				CPURequests:    "30m",
+				MemoryRequests: "25Mi",
+				CPULimits:      "200m",
+				MemoryLimits:   "512Mi",
```

Comment:
These limits numbers seem high, is that what's being used in practice?

Reply:
which one is higher cpu or memory?

Reply:
Both, but memory more so.

These are sidecar "health check" style bash scripts that run as daemonsets, right? I think using up a half-gig of memory in particular as required overhead, per-node, just to use Azure CNI, seems wrong. @mboersma @devigned  thoughts?

Reply:
yes this daemonset would not consume 512M. I just gave that as limit but potentially it should not hit that. Same goes for cpu

Reply:
The purpose of a resource limit is to define the upper bound of a container beyond which a pod containing it may be terminated (depending on the density of the node on which the pod is scheduled). Let's set the limit to that number. It should not be, I don't think, over an order of magnitude greater than the expected working footprint (in this case `25Mi`). A good rule of thumb is to define your requests with a little bit of overhead beyond the stable state that your stress tests prove are the norm (say, 2x); and then ideally your limits value would be identical. Otherwise the configuration signals that the underlying runtime is not stable.

Again, would love thoughts from @devigned and @mboersma. This particular runtime doesn't sound like one that needs a lot of tolerance for memory peaks and valleys, and so setting a higher limit would (1) prevent optimal node pod scheduling density for folks running Azure CNI (they may have to absorb 512Mi *just* for a health check) and (2) obfuscate memory leaks in the underlying code as a result of excessive tolerance.

Also cc @CecileRobertMichon @marosset @ritazh

Reply:
I like the suggestion of 2x more than stable state as general guidance. Just keep it as low as reasonable and be able to recover if the container gets killed for going over that limit.